### PR TITLE
Improvements to Jenkinsfile

### PR DESCRIPTION
**Pull request by justaddcoffee**
- wrap stages in dir('gitrepo') instead of repeatedly doing `cd gitrepo`
- upload output of transform() to s3
- fix logic in download() that prevents proper fxn in non-master branch

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -43,57 +45,72 @@ pipeline {
...
             }
         }
+
         stage('Load') {
             steps {
-                sh 'cd config;. venv/bin/activate; python3.7 run.py load'
-                sh 'cd config;. venv/bin/activate; pigz merged-kg.tar'
+                dir('./gitrepo') {
+                    sh '. venv/bin/activate; python3.7 run.py load'
+                    sh '. venv/bin/activate; pigz merged-kg.tar'
```
Comment (kltm) [inline Jenkinsfile:81]: Why does this need activate?

File: Jenkinsfile [hunk 2]
```diff
@@ -43,57 +45,72 @@ pipeline {
...
             }
         }
+
         stage('Convert to RDF') {
             steps {
-                sh 'cd config;. venv/bin/activate; kgx transform --input-type tsv --output-type nt -o ./merged-kg.nt merged-kg.tar.gz'
-                sh 'cd config;. venv/bin/activate; pigz merged-kg.nt'
+                dir('./gitrepo') {
+                    sh '. venv/bin/activate; kgx transform --input-type tsv --output-type nt -o ./merged-kg.nt merged-kg.tar.gz'
+                    sh '. venv/bin/activate; pigz merged-kg.nt'
```
Comment (kltm) [inline Jenkinsfile:90]: When is activate used?

File: Jenkinsfile [hunk 3]
```diff
@@ -43,57 +45,72 @@ pipeline {
...
+                dir('./gitrepo') {
+                    script {
+                        def run_py_dl = sh(
+                            script: '. venv/bin/activate; python3.7 run.py download', returnStatus: true
+                        )
+                        withCredentials([file(credentialsId: 's3cmd_kg_hub_push_configuration', variable: 'S3CMD_JSON')]) {
+                            if (run_py_dl == 0) { // upload raw to s3
+                                sh 's3cmd -c $S3CMD_JSON --acl-public --mime-type=plain/text --cf-invalidate put -r data/raw s3://kg-hub-public-data/'
+                            } else { // 'run.py download' failed - let's try to download last good copy of raw/ from s3 to data/
+                                sh 'rm -fr data/raw; mkdir -p data/raw'
```
Comment (kltm) [inline Jenkinsfile:60]: I might suggest:

'rm -r -f data/raw || true'
'mkdir -p data/raw || true'

or functionally similar

File: Jenkinsfile [hunk 4]
```diff
@@ -43,57 +45,72 @@ pipeline {
...
+                }
             }
         }
+
         stage('Load') {
             steps {
-                sh 'cd config;. venv/bin/activate; python3.7 run.py load'
-                sh 'cd config;. venv/bin/activate; pigz merged-kg.tar'
+                dir('./gitrepo') {
+                    sh '. venv/bin/activate; python3.7 run.py load'
```
Comment (kltm) [inline Jenkinsfile:80]: maybe '&&' for these?

File: Jenkinsfile (hunk 1 above)
Comment (justaddcoffee) [reply Jenkinsfile:81]: `run.py load` needs the venv environment - I guess `pigz` doesn't though

File: Jenkinsfile (hunk 2 above)
Comment (justaddcoffee) [reply Jenkinsfile:90]: all the `run.py` commands will need this, possibly kgx too. Looks like I can remove it from the pigz command though

File: Jenkinsfile (hunk 3 above)
Comment (justaddcoffee) [reply Jenkinsfile:60]: thanks

File: Jenkinsfile (hunk 4 above)
Comment (justaddcoffee) [reply Jenkinsfile:80]: probably a good idea

**Pull request merged**